data/src/main/java/com/moonpig/data/ProductDetailStubRepository.kt

```diff
@@ -5,34 +5,41 @@ import com.moonpig.mvisample.domain.productdetail.AddProductRequest
 import com.moonpig.mvisample.domain.productdetail.ProductDetailRepository
 import com.moonpig.mvisample.domain.productdetail.RepositoryState
 import io.reactivex.Observable
+import java.util.concurrent.TimeUnit
 
 class ProductDetailStubRepository : ProductDetailRepository {
     override fun getProductDetails(productId: String): Observable<RepositoryState.GetProductDetail> =
             when (productId) {
-                "1" -> respondWithCatArmor()
+                catArmourProductId -> respondWithCatArmour()
                 else -> respondWithLoadError()
             }
 
-    private fun respondWithCatArmor() =
-            Observable.just(
-                    RepositoryState.GetProductDetail.InFlight,
-                    RepositoryState.GetProductDetail.Success(
-                            ProductDetail(
-                                    "Advanced Cat Armor",
-                                    "Everything your kitty needs to seriously up it's fighting game! This stylish armor will protect your cat from ordinary piercing and blunt damage. Accessories including the pictured pole not included.",
-                                    299,
-                                    "https://i.postimg.cc/QtwNZdm4/cat-armor-1.jpg"
+    private fun respondWithCatArmour() =
+            Observable.merge(
+                    Observable.just(RepositoryState.GetProductDetail.InFlight),
+                    Observable.just(
+                            RepositoryState.GetProductDetail.Success(
+                                    ProductDetail(
+                                            "Advanced Cat Armour",
+                                            "Everything your kitty needs to seriously up it's fighting game! This stylish armour will protect your cat from ordinary piercing and blunt damage. Accessories including the pictured pole not included.",
+                                            299,
+                                            "https://i.postimg.cc/QtwNZdm4/cat-armor-1.jpg"
+                                    )
                             )
-                    )
+                    ).delay(1, TimeUnit.SECONDS)
             )
 
     private fun respondWithLoadError() =
-            Observable.just(
-                    RepositoryState.GetProductDetail.InFlight,
-                    RepositoryState.GetProductDetail.Error(RuntimeException())
+            Observable.merge(
+                    Observable.just(RepositoryState.GetProductDetail.InFlight),
+                    Observable.just(RepositoryState.GetProductDetail.Error(RuntimeException())).delay(2, TimeUnit.SECONDS)
```

Comment:
Test for this?

Reply:
There were no tests for the stub repository when I started working on it. 
Would you like me to add some? 

Could get a bit tricky with the delay

Reply:
Ah weren't there? Ok no don't bother